samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -290,23 +286,25 @@ public void unsubscribeChildChanges(String path, IZkChildListener childListener)
 
   public void writeData(String path, Object object) {
     zkClient.writeData(path, object);
-    if (metrics != null) {
-      metrics.writes.inc();
-    }
+    metrics.writes.inc();
   }
 
   public boolean exists(String path) {
     return zkClient.exists(path);
   }
 
   public void close() throws ZkInterruptedException {
-    try {
-      zkClient.close();
-    } catch (ZkInterruptedException e) {
-      // Swallowing due to occurrence in the last stage of lifecycle (Not actionable) and clear the interrupted status.
-      Thread.interrupted();
-      LOG.warn("Ignoring the exception when closing the zookeeper client.", e);
-    }
+    int retryCount = 0;
+    do {
+      try {
+        zkClient.close();
+        return;
+      } catch (ZkInterruptedException e) {
+        Thread.interrupted();
```

Comment:
We shouldn't be clearing the interrupt status of the thread. Not restoring interrupt can often be dangerous since it masks the existence of an interrupt upstream in the current-thread's call-stack.  If the current thread is interrupted, we should be restoring its interruption status back right?